File: (7 kyu) Even numbers in an array/(7 kyu) Even numbers in an array.rs
fn even_numbers(array: Vec<i32>, number: usize) -> Vec<i32> {
  let arr = array.into_iter().filter(|&i|i % 2 == 0).collect::<Vec<_>>();
  let from = arr.len() - number;
  arr[from..].to_vec()
}

File: (7 kyu) Even numbers in an array/(7 kyu) Even numbers in an array.cpp
#include <vector>

using namespace std;

std::vector<int> evenNumbers(vector<int> arr, int n)
{
    vector<int> result;

    for (auto x : arr)
        if (x % 2 == 0)
            result.push_back(x);

    return vector<int>(result.end() - n, result.end());
}